### PR TITLE
Support QR code Model 1 and some other chores

### LINKED ISSUE #390: Component types not included in library
Even though this library is made using typescript, and types are included for the `misc` package, types are not included for the components themselves, which show up as `any`.

image

### PR DESCRIPTION
**Pull request by Sec-ant**
- Update [`barcode-detector`](https://www.npmjs.com/package/barcode-detector) to v2.1.0 to support QR code Model 1, as it is just supported by zxing-cpp: https://github.com/zxing-cpp/zxing-cpp/pull/633.
- [`@sec-ant/zxing-wasm`](https://www.npmjs.com/package/@sec-ant/zxing-wasm) is refactored and republished as [`zxing-wasm`](https://www.npmjs.com/package/zxing-wasm). The `.wasm` file urls are therefore also changed.
- Export type `BarcodeFormat`.
- Update dependencies and fix #390.

image

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -24,8 +24,13 @@ export default defineConfig({
...
     lib: {
-      entry: resolve(__dirname, 'src/index.ts'),
-      name: 'VueQrcodeReader'
+      entry: {
+        'vue-qrcode-reader': resolve(__dirname, 'src/index.ts')
+      },
+      formats: ['es', 'umd'],
+      name: 'VueQrcodeReader',
+      fileName: (format, entryName) =>
+        format === 'es' ? `${entryName}.js` : `${entryName}.${format}.js`
```
Comment (gruhn) [inline vite.config.ts:33]: Thanks for finding a proper solution to my cheap-ass workaround in the `package.json`. However, that means we don't have a `dist/vue-qrcode-reader.umd.cjs` anymore, right? I guess strictly speaking this breaks backwards compatibility. Is it possible to also generate a copy of `dist/vue-qrcode-reader.umd.js` that is named `dist/vue-qrcode-reader.umd.cjs` ?

File: README.md [hunk 2]
```diff
@@ -164,7 +164,7 @@ Use kebab-case to reference them in your templates:
 #### Some of my QR codes are not being detected.
 
 - Make sure, there is some white border around the QR code.
-- [Model 1 QR codes](https://en.wikipedia.org/wiki/QR_code#Model_1) are not supported.
+- Test your QR codes in the upstream packages: [`barcode-detector`](https://github.com/Sec-ant/barcode-detector) -> [`zxing-wasm`](https://github.com/Sec-ant/zxing-wasm) -> [`zxing-cpp`](https://github.com/zxing-cpp/zxing-cpp), and file detection issues in the highest-level repository where the problem first occurs.
```
Comment (gruhn) [inline README.md:167]: Damn, you even updated this :+1:

File: vite.config.ts (hunk 1 above)
Comment (Sec-ant) [reply vite.config.ts:33]: It shouldn't break any things if users are using pinned versions. And jsdelivr (or maybe other CDNs too) seems to have a fallback mechanism when a file is no longer served in the upgraded version.

Comment (gruhn) [reply vite.config.ts:33]: Ok yeah that's good enough for this edge case :+1:

**Pull request merged**